src/server/rdb_save.cc

```diff
@@ -129,7 +129,7 @@ uint8_t RdbObjectType(unsigned type, unsigned compact_enc) {
       break;
     case OBJ_HASH:
       if (compact_enc == kEncodingListPack)
-        return RDB_TYPE_HASH_ZIPLIST;
+        return RDB_TYPE_HASH_LISTPACK;
```

Comment:
That was a bug, we use listpack encoding at runtime and we serialized it as `RDB_TYPE_HASH_ZIPLIST`

Reply:
nope it was for compatibility as you said later on. I revert this. Once we decide to serialize the way redis 7 does, we can bring the change from here.